vllm/spec_decode/spec_decode_worker.py

```diff
@@ -461,6 +471,12 @@ def _run_speculative_decoding_step(
         proposals = self.proposer_worker.get_spec_proposals(
             execute_model_req, self._seq_with_bonus_token_in_last_step)
 
+        if not self.allow_no_draft_tokens and sum(
+                proposals.proposal_lens) == 0:
```

Comment:
I'm a bit worry about the overhead that `sum` brings, but I feel it's fine for now given that it won't be triggered with draft model TP=1. wdyt @cadedaniel?

Reply:
yeah, is it possible to store this field in `proposals` when we create `proposals`? that way we don't need an additional CPU-GPU-CPU sync